dart/lib/src/transport/http_transport.dart

```diff
@@ -19,19 +22,22 @@ class HttpTransport implements Transport {
 
   final RateLimiter _rateLimiter;
 
+  final ClientReportRecorder _recorder;
+
   late _CredentialBuilder _credentialBuilder;
 
   final Map<String, String> _headers;
 
-  factory HttpTransport(SentryOptions options, RateLimiter rateLimiter) {
+  factory HttpTransport(SentryOptions options, RateLimiter rateLimiter,
+      ClientReportRecorder recorder) {
```

Comment:
If the options already have the `recorder`, we don't need the recorder as ctor param.